### PR TITLE
[custom response filter] Prevent overriding the status code with 100

### PR DESCRIPTION
**Pull request by pradeepcrao**
Commit Message:
Additional Description:
Risk Level: Low
Testing: ran //test/... and built docs
Docs Changes: N/A
Release Notes: Added release note

### CODE REVIEW

File: changelogs/current.yaml [hunk 1]
```diff
@@ -147,6 +147,10 @@ minor_behavior_changes:
     config. Previously, the config only removes JWT if set in headers. With this addition, the config can also be
     used to remove JWT set in query parameters. This behavior can be reverted by setting the runtime guard
     ``envoy.reloadable_features.jwt_authn_remove_jwt_from_query_params`` to false.
+- area: custom response filter
+  change: |
+    Prevent the redirect  policy from sending 100 as the overridden response code as this does not make sense
```
Comment (adisuissa) [inline changelogs/current.yaml:152]: nit:
```suggestion
    Prevent the redirect policy from sending 100 as the overridden response code as the custom error response must be complete.
```

**Review by adisuissa — COMMENTED** [review]
Thanks!
Left a small nit.
Technically it may break some configs, but I agree that sending 1xx response in this case should be disabled.

File: changelogs/current.yaml (hunk 1 above)
Comment (pradeepcrao) [reply changelogs/current.yaml]: Thanks for the review Adi! Done.

File: changelogs/current.yaml [hunk 2]
```diff
@@ -147,6 +147,10 @@ minor_behavior_changes:
     config. Previously, the config only removes JWT if set in headers. With this addition, the config can also be
     used to remove JWT set in query parameters. This behavior can be reverted by setting the runtime guard
     ``envoy.reloadable_features.jwt_authn_remove_jwt_from_query_params`` to false.
+- area: custom response filter
+  change: |
+    Prevent the redirect  policy from sending 100 as the overridden response code as the custom response must
```
Comment (adisuissa) [inline changelogs/current.yaml:152]: apologies for the nitpick:
```suggestion
    Prevent the redirect policy from sending 100 as the overridden response code as the custom response must
```

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!
/lgtm api

**Pull request merged**